crates/prover/src/components/memory.rs

```diff
@@ -19,10 +17,14 @@ use stwo_prover::core::pcs::TreeVec;
 use stwo_prover::core::poly::circle::{CanonicCoset, CircleEvaluation};
 use stwo_prover::core::poly::BitReversedOrder;
 
+use crate::adapter::memory::MemoryBoundaries;
 use crate::relations;
 
-#[derive(Copy, Clone, Default)]
+const N_M31_IN_MEMORY_ENTRY: usize = 7;
```

Comment:
right now the struct MemoryEntry has only 5 m31 afaik
maybe we mean N_M31_IN_MEMORY_PER_ROW?

there is something i dont understand: `addr, value (4), prev_value, new_value, clk, prev_clk` is 9 values, what did i miss?

Reply:
this is the memory component for initial and final values, ie boundary conditions. So it's just a list of entries with appropriate multiplicity (-1 or +1)